lib/plugins/expressionParser.js

```diff
@@ -177,7 +177,7 @@ function expressionApplier(context, typeInformation) {
         };
 
         /*jshint camelcase: false */
-        if (config.checkNgsi2() && attribute.object_id) {
+        if (config.checkNgsi2() || config.checkNgsiLD() && attribute.object_id) {
```

Comment:
```suggestion
        if ((config.checkNgsi2() || config.checkNgsiLD()) && attribute.object_id) {
```

We think is more explicit and clear this way. What do you think @jason-fox ?

Reply:
I agree.

Reply:
Fixed in 1aedb33

Reply:
Thinking in the previous implementation before the fix, it seems it is more than an stetic improvements, it is an actual fix given that:

```
> true || false && false
true

> (true || false) && false
false
```

Maybe some unit test is missing? (otherwise the problem should have raised at unit test stage at travis...) Could you added, please? (as a PR to branch feature/842_ngsi_ld)

Reply:
Looking at this in more detail, I can't see that adding a test case makes sense here. Yes the logic was wrong, but the assign statement had no impact. 

```javascript
 if ((config.checkNgsi2() || config.checkNgsiLD()) && attribute.object_id) {
            newAttribute.object_id = attribute.object_id;
} 
```

will mistakenly attempt to set  `newAttribute.object_id`  without those brackets right?  However `attribute.object_id` ain't` false`, it is `undefined` and `newAttribute.object_id = undefined`  won't set the attribute, so the payload remains the same.

So overall the effect of the bug was to unset an internal variable which hadn't been set in the first place. No wonder there wasn't a test for it.

The only things which _may_ make sense to alter:

#### Fail faster - move the attribute falsy test to the start of the line.

```javascript
 if (attribute.object_id && (config.checkNgsi2() || config.checkNgsiLD()) {
            newAttribute.object_id = attribute.object_id;
} 
```

####  Remove the undefined check as unnecessary.

```javascript
 if (config.checkNgsi2() || config.checkNgsiLD()) {
            newAttribute.object_id = attribute.object_id;
} 
```

What do you think?